### ISSUE TITLE
BookLibConnect Continually Terminates Download

### ISSUE THREAD

**Issue by Sailfish** [audiamus/BookLibConnect#108]
Every time I start BookLibConnect it terminates the download at the Black Talon title. This title was an Audible Plus book I added to my library Back in May and was 1487 in my collection. My current title count now stands at 1506. In further research, I noticed that the BookLibConnect img folder stops loading at in the 1480ish range (sometimes 1487, 1483, &c) but always shows the display list stopping with the Black Talon title at the top. I even removed the title from my library but it doesn't seem to make a difference.

I've also attached the  -Log=4 run of the log file.

[BookLibConnect_2023-07-21_001.log](https://github.com/audiamus/BookLibConnect/files/12135630/BookLibConnect_2023-07-21_001.log)
![BookLibConnect_img_folder](https://github.com/audiamus/BookLibConnect/assets/110050/3786c544-27e0-42bf-afbf-aad95b091e8d)
![BookLibConnectBlackTalon](https://github.com/audiamus/BookLibConnect/assets/110050/3bc12811-4c94-469e-977e-da8de19da7ff)

**Comment by audiamus** [comment]
It seems that you get a "license denied" response for part 1 of the book, but a "license granted" for part 2. Have you tried to download as a single part (disable "Multiple-part download" in the settings)?

**Comment by Sailfish** [comment]
There may be a misunderstanding on my part, I'm unable to download title that are added to my library by Audible Plus feature as I'm only able to play them via web site or smartphone Audible App.

**Comment by audiamus** [comment]
Well, in the latest log, the update library request to Audible timed out. You should try again later. Nonetheless, your settings still say that you are downloading in multi-part mode.

**Comment by Sailfish** [comment]
I missed deselecting the multi-part mode checkbox, thank you for your patience. Attached is the latest log from BookLibConnect with the box de-selected. It continues to appear to prematurely terminate at that same point. Have you other users with library titles as large or larger than mine? Is there a possibility that size of my library might be exceeding some limit within BookLibConnect? 
[BookLibConnect_2023-07-22_001.log](https://github.com/audiamus/BookLibConnect/files/12137189/BookLibConnect_2023-07-22_001.log)

Regards,
Pat

**Comment by audiamus** [comment]
You still encounter a timeout connecting the server to update your library. No idea what the cause is. You could try next time with "Auto Connect" enabled. That will refresh your token. (Other developers say the token usually does not expire but I am not sure.)

**Comment by Sailfish** [comment]
Nailed It! After enabling Auto Connect it worked at once and again, after clearing it out and trying it again.

Relatedly, I recall the reason I had enabled Multi-part download. I had a large amount of .aa titles from before .aax format became the norm and I thought that without it enabled, BookLibConnect may have ignored the .aa multi-part titles and may have been the cause of short title counts. After enabling Auto connect, I confirmed that all of titles have been downloaded. I suspect it may be because Audible had since up-scaled all of their prior .aa titles to the .aax format.

As always, your prompt and thorough support is unique and superior to other products in the industry, including those that charge for them.

I have another Issue but will address it in another Issue post.
